### ISSUE TITLE
New to find a new steward

### ISSUE THREAD

**Issue by warsaw** [aio-libs/aiosmtpd#144]
I think it's time for me to admit that I don't have much time to work on aiosmtpd.  I really hope it doesn't languish though, given how useful of a library it is.

We need to identify someone else to do most of the project management for this library.  Any takers?

**Comment by warsaw** [comment]
I'm glad you find this useful!  I don't think it will take a lot of time, but it takes more consistency and diligence to keep up on the PRs and issues than I have available right now.  Are you interested in reviewing the PRs and issues?

**Comment by waynew** [comment]
So, my life was insanely busy for the past couple of months, but I think that I've reached a point where I could spend a couple of hours each week reviewing PRs and issues :)

**Comment by warsaw** [comment]
Awesome, thanks @waynew !

**Comment by asvetlov** [comment]
My spare time is limited but you can consider me as a backup person.

**Comment by pepoluan** [comment]
I'd like to volunteer some of my time, if the offer still stands.

**Comment by asvetlov** [comment]
@pepoluan invitation is sent

**Comment by pepoluan** [comment]
> @pepoluan invitation is sent

Thank you! I have accepted the invitation.

**Comment by pepoluan** [comment]
So, do we have a list of tech debts to do?

**Comment by asvetlov** [comment]
There are pending pull requests and open issues

**Comment by pepoluan** [comment]
Re: Title Change

No I'm not retiring from this project, but the thread name _was_ a bit ... awkward 😄

**Comment by pepoluan** [comment]
@asvetlov @warsaw @waynew can you add me as "Maintainer" to https://pypi.org/project/aiosmtpd/ so I can upload new releases?

**Comment by warsaw** [comment]
@pepoluan Done!  Your invite should be pending.

**Comment by pepoluan** [comment]
> @pepoluan Done! Your invite should be pending.

Thank you! I have accepted the invitation.

**Comment by pepoluan** [comment]
If y'all noticed that my output is a bit ... slow lately, fear not. I'm not burnt out and/or losing my interest in the project, I'm just... "transitioning" from "employed" status to "free agent"; my company, a startup, ran out of cash and now I'm handing over _everything_ to the main investor and it had been taking a LOT of my time & energy.

When everything's done, I'll get back to my previous level of productivity.

**Comment by diazona** [comment]
Hi everyone! If I may ask, what's the current status of this? Since this issue is still featured and there have been no commits in more than a year, it looks to a casual observer (i.e. me) as though the package is unmaintained. But @pepoluan I see your comment about being busy with a career transition, and I don't know if that still applies and you intend to resume activity on this package.

For context, I'm looking to use `aiosmtpd` as a replacement for `smtpd` [in a package I maintain](https://github.com/pytest-dev/pytest-localserver/pull/30), but I have some reservations about doing so if `aiosmtpd` is not actively developed.

**Comment by warsaw** [comment]
> For context, I'm looking to use aiosmtpd as a replacement for smtpd https://github.com/pytest-dev/pytest-localserver/pull/30, but I have some reservations about doing so if aiosmtpd is not actively developed.

You also have to consider that [smtpd is being deprecated and will be removed in Python 3.12](https://peps.python.org/pep-0594/#deprecated-modules) so it won't even be an option for you in the future.

**Comment by diazona** [comment]
Sure, understood. In this case, if `aiosmtpd` is not being maintained, I might wind up removing SMTP support from `pytest-localserver` entirely at some point.

**Comment by waynew** [comment]
That's a fair criticism. I certainly haven't been able to devote as much time to the project as I would like. Like most other OS projects (and other efforts in life) time is the main limiting factor. I find personally I also struggle with the time to spin back up - there's several whole RFCs one has to renew familiarity with when reviewing PRs.

**Comment by warsaw** [comment]
@waynew Are there specific feature requests or bugs you need addressed?  How much maintenance does this library need?

**Comment by waynew** [comment]
I meant more to review the outstanding PRs :upside_down_face: Mostly it *shouldn't* need a lot of maintenance (aside from supporting latest & greatest Pythons, of course), as the SMTP specs are pretty well defined.

**Comment by diazona** [comment]
Yeah, as long as somebody is able to fix (or review/accept PRs to fix) any significant bugs or any compatibility issues with future versions of Python that may come up, that's good enough for me, and probably for most other projects that depend on aiosmtpd. :+1: Basically what I'm trying to find out by posting here is whether someone will be performing that role for the foreseeable future.

BTW just in case there's anything I can do to help, I'd be happy to do what I can, but I don't have a whole lot of experience in asynchronous programming nor much free time these days, so I suspect my options are pretty limited.

**Comment by nikochiko** [comment]
Hi, I have found aiosmtpd to be very useful for me. I'm willing to put in some time each week to maintainence of this project, if another hand is needed.

**Comment by waynew** [comment]
@nikochiko that would be super helpful! Personally my main challenge is that coming back to an issue takes me a while to spin the information back up into my mind. I struggle to keep a consistent schedule with that.

**Comment by nikochiko** [comment]
Sure, I understand. I'm happy to help.
I'm getting started with the setup. I'll push a fix for updating the supported Pythons, and then take a look at some of the bugs. Really happy to do this :)

**Comment by webknjaz** [comment]
Hey @matrixise, would you be interested in getting involved? I saw your `dsmtpd` depends on this project now, and you're probably a good candidate for getting a backup commit bit.

**Comment by matrixise** [comment]
Hey @webknjaz, thanks for thinking of me!

I'm happy to accept a backup commit bit. I should be honest though: I don't have much bandwidth to commit to regular maintenance. I'll keep an eye on things from time to time and help out where I can — especially for issues that affect `dsmtpd` — but I wouldn't want you to count on me as a primary reviewer.

If that works for you, I'm in!

**Comment by webknjaz** [comment]
@matrixise there's currently one external contributor trying to address the maintenance debt and no active primary maintainer. I've been helping out with reviewing the infra changes, but that's about it. So just having you additionally would be useful regardless.

**Comment by matrixise** [comment]
Sounds good! I'll take it seriously — even if my availability is limited, I'll make sure to review things carefully when I do. If there's a contributing guide or any process to follow for maintainers, feel free to point me to it.

**Comment by webknjaz** [comment]
Ack!

**Comment by webknjaz** [comment]
@matrixise I've sent out an invitation.

**Comment by matrixise** [comment]
Thanks @webknjaz for the invitation! I've accepted it. I hope I'll be a good contributor and that things will go well going forward.
